cmd/sonobuoy/app/run.go

```diff
@@ -83,10 +85,23 @@ func submitSonobuoyRun(f *genFlags) func(cmd *cobra.Command, args []string) {
 		}
 
 		if !f.skipPreflight {
+			preflightChecks := make(map[string]func(kubernetes.Interface, *client.PreflightConfig) error)
+
+			for key, value := range sbc.GetValidPreflightChecks() {
+				preflightChecks[key] = value
+			}
+
+			for _, name := range f.skipPreflightChecks {
+				if _, ok := sbc.GetValidPreflightChecks()[name]; ok {
```

Comment:
My other comment below talks about thinking this would be cleaner if you push this kind of logic down closer to where the preflight checks actually get launched, and in that situation you'll have the original `validPreflightChecks` funcMap to check against instead of calling to he method to get a reference to it.

Handle the `!ok` case as well so users are informed if they provided a nonsense value. You may want to normalize the input a bit too, just make everything `strings.toLower` to avoid tiny issues like capitalization from mattering/causing a user to error.

Reply:
Moved logic to preflight.go